tests/main.rs

```diff
@@ -180,143 +180,145 @@ fn find_results(
-
-                let binary_run = Command::new(&exe_name)
-                    .current_dir(&path.parent().unwrap())
-                    .output()?;
-
-                let binary_output = String::from_utf8_lossy(&binary_run.stdout).to_string();
-                let binary_output = binary_output.replace("\r\n", "\n");
-
-                let binary_stderr_output = String::from_utf8_lossy(&binary_run.stderr).to_string();
-                let binary_stderr_output = binary_stderr_output.replace("\r\n", "\n");
-
-                let baseline_text = expected_output;
-                let baseline_text = baseline_text.map(|text| text.replace("\r\n", "\n"));
-
-                let baseline_stderr_text = expected_stderr;
-                let baseline_stderr_text =
-                    baseline_stderr_text.map(|text| text.replace("\r\n", "\n"));
-
-                let mut stderr_checked = false;
-                if let Some(baseline_stderr_text) = baseline_stderr_text {
-                    assert!(
-                        // Because runtime assertion text depends on file name, C++ library line names etc., the exact stderr text is both unpredictable and volatile.
-                        // Therefore, stderr checks allow to only check for some substring of the actual output,
-                        // where we can e.g. put in the exact assertion text we are checking for.
-                        // This is still vulnerable to changes in the assertion expression, but we do need to check for something.
-                        // It looks like the best solution right now.
-                        binary_stderr_output.contains(&baseline_stderr_text),
-                        "\r\n Test (stderr): {}",
-                        path.to_string_lossy()
-                    );
-                    stderr_checked = true;
-                }
+    // If we checked stderr, we're not required to check stdout.
+    if !stderr_checked && baseline_text.is_some() {
+        assert_eq!(
+            binary_output,
+            baseline_text.unwrap(),
+            "\r\nTest: {}",
+            path.to_string_lossy()
+        );
+    }
 
-                // If we checked stderr, we're not required to check stdout.
-                if !stderr_checked && baseline_text.is_some() {
-                    assert_eq!(
-                        binary_output,
-                        baseline_text.unwrap(),
-                        "\r\nTest: {}",
-                        path.to_string_lossy()
-                    );
-                }
+    Ok(())
+}
+
+fn test_samples(path: &str) -> Result<(), JaktError> {
+    for sample in std::fs::read_dir(path).expect("if this fails, the repo or harness is broken") {
+        let sample = sample.expect("if this fails, the filesytem is having issues");
+
+        let path = sample.path();
+
+        if let Some(ext) = path.extension() {
+            if ext == "jakt" {
+                test_sample(&path);
```

Comment:
We're getting warnings on this line that the result is being ignored.

Reply:
Thanks for the teminder forgot to handle that, will update the PR later today.